rest/route/sns.go

```diff
@@ -292,9 +307,85 @@ func (aws *awsSns) handleInstanceStopped(ctx context.Context, instanceID string)
 		return nil
 	}
 
-	if err := amboy.EnqueueUniqueJob(ctx, aws.queue, units.NewHostMonitorExternalStateJob(aws.env, h, aws.payload.MessageId)); err != nil {
+	if err := amboy.EnqueueUniqueJob(ctx, sns.queue, units.NewHostMonitorExternalStateJob(sns.env, h, sns.payload.MessageId)); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+type ecsSNS struct {
+	baseSNS
+}
+
+type ecsEventBridgeNotification struct {
+	DetailType string         `json:"detail-type"`
+	Detail     ecsEventDetail `json:"detail"`
+}
+
+type ecsEventDetail struct {
+	TaskARN    string `json:"taskArn"`
+	LastStatus string `json:"lastStatus"`
+}
+
+func makeECSSNS(sc data.Connector, env evergreen.Environment, queue amboy.Queue) gimlet.RouteHandler {
+	return &ecsSNS{
+		baseSNS: makeBaseSNS(sc, env, queue),
+	}
+}
+
+func (sns *ecsSNS) Factory() gimlet.RouteHandler {
+	return &ecsSNS{
+		baseSNS: makeBaseSNS(sns.sc, sns.env, sns.queue),
+	}
+}
+
+func (sns *ecsSNS) Run(ctx context.Context) gimlet.Responder {
+	// Subscription/Unsubscription is a rare action that we will handle manually
+	// and will be logged to splunk given the logging level.
+	switch sns.messageType {
+	case messageTypeSubscriptionConfirmation:
+		grip.Alert(message.Fields{
+			"message":       "got AWS SNS subscription confirmation. Visit subscribe_url to confirm",
+			"subscribe_url": sns.payload.SubscribeURL,
+			"topic_arn":     sns.payload.TopicArn,
+		})
+	case messageTypeUnsubscribeConfirmation:
+		grip.Alert(message.Fields{
+			"message":         "got AWS SNS unsubscription confirmation. Visit unsubscribe_url to confirm",
+			"unsubscribe_url": sns.payload.SubscribeURL,
+			"topic_arn":       sns.payload.TopicArn,
+		})
+	case messageTypeNotification:
+		// TODO (EVG-14811): handle SNS notification for ECS task state change
+		// and remove this debug log.
+		grip.Info(message.Fields{
+			"message":      "received ECS SNS notification",
+			"message_id":   sns.payload.MessageId,
+			"message_type": sns.payload.Type,
+			"topic":        sns.payload.TopicArn,
+		})
+	default:
+		grip.Error(message.Fields{
+			"message":         "got an unknown message type",
+			"type":            sns.messageType,
+			"payload_subject": sns.payload.Subject,
+			"payload_message": sns.payload.Message,
+			"payload_topic":   sns.payload.TopicArn,
+		})
+		return gimlet.NewTextErrorResponse(fmt.Sprintf("message type '%s' is not recognized", sns.messageType))
+	}
+
+	return gimlet.NewJSONResponse(struct{}{})
+}
+
+func (sns *ecsSNS) handleNotification(ctx context.Context) error {
```

Comment:
Can we add a TODO here?

Reply:
There's a TODO in L360.

Reply:
I know, but I think this warrants another one since this function is incomplete.

Reply:
Done.